library/core/src/slice/mod.rs

```diff
@@ -865,8 +865,9 @@ impl<T> [T] {
     pub fn chunks_exact(&self, chunk_size: usize) -> ChunksExact<'_, T> {
         assert_ne!(chunk_size, 0);
         let rem = self.len() % chunk_size;
-        let len = self.len() - rem;
-        let (fst, snd) = self.split_at(len);
+        let fst_len = self.len() - rem;
+        // SAFETY: 0 <= fst_len < self.len() by construction above
```

Comment:
when `self.len() == 0` here and other places:
```suggestion
        // SAFETY: 0 <= fst_len < self.len() by construction above
```

Reply:
Indeed, thanks